### DISCUSSION TITLE
Include 'Allow' header on default 405 Method Not Allowed responses

### DISCUSSION THREAD

**Discussion by hellocoldworld** [fastapi/fastapi#6505] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.testclient import TestClient

app = FastAPI()


@app.get("/")
def read_root():
    return {"Hello": "World"}


client = TestClient(app)


def test_method_not_allowed_has_allow_headers():
    response = client.put("/")
    assert response.status_code == 405, response.status_code
    assert "allow" in response.headers, response.headers
```

### Description

A 405 Method Not Allowed response should include an 'Allow' header indicating what mehtods are allowed (ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405)
This was already pointed out in https://github.com/tiangolo/fastapi/issues/1773#issuecomment-678289598 but since that issue and the related PR's seems stalled, I thought it could be good opening a dedicated issue.

### Wanted Solution

The 'allow' header should be included with a list of allowed methods

### Wanted Code

```python
from fastapi import FastAPI
from fastapi.testclient import TestClient

app = FastAPI()


@app.get("/")
def read_root():
    return {"Hello": "World"}


client = TestClient(app)


def test_method_not_allowed_has_allow_headers():
    response = client.put("/")
    assert response.status_code == 405, response.status_code
    assert "allow" in response.headers, response.headers
    assert response.headers["allow"] == "GET, HEAD"
```

### Operating System

Linux

### FastAPI Version

0.70.1

### Python Version

Python 3.9.6

**Comment by Kludex** [comment]
We'll handle this on Starlette's side. Thanks for the issue.

**Comment by Kludex** [comment, marked as answer]
It will be available on the next Starlette release, and consequently in some FastAPI.

Related PR: https://github.com/encode/starlette/pull/1436

**Comment by hellocoldworld** [comment]
Fixed on Starlette side ⚡
